spec/support/container_helpers.rb

```diff
@@ -0,0 +1,123 @@
+require 'stringio'
+
+# rubocop:disable Metrics/ModuleLength
+module ContainerHelpers
+  shared_context 'cgroup file' do
+    let(:cgroup_file) { StringIO.new }
+
+    before do
+      allow(File).to receive(:open).and_call_original
+      allow(File).to receive(:open)
+        .with('/proc/self/cgroup')
+        .and_return(cgroup_file)
+    end
+  end
+
+  shared_context 'non-containerized environment' do
```

Comment:
We should also test the case when the file does not exist (e.g. windows or something).

Reply:
Sure, that's a good thought. We can do that in the `cgroup_spec.rb` tests.

Reply:
Added specs for this.